EIPS/eip-hard-gas-target.md

```diff
@@ -0,0 +1,60 @@
+---
+eip: xxxx
+title: Hardcoded Block Gas Target
+author: Philippe Castonguay (@PhABC)
+discussions-to: xxxx
+status: Draft
+type: Standards Track
+category: Core
+created: 2020-03-13
+
+---
+
+## Simple Summary
+
+Hardcode the block gas target (previously known as *block gas limit* before [EIP-1559](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1559.md)) to `12,500,000` gas per block.
+
+## Abstract
+
+Ethereum's block gas target is currently dictated by block producers and is not enforced when validating blocks in clients. This EIP proposes to hardcode the block gas target to `12,500,000`.
+
+## Motivation
+
+Both Ethereum's Proof of Work and Proof of Stake designs assume that block producers are financially rational, but does not assume block producers to be benevolent. There is one exception however, and it is when block producers choose the gas targe of a block (previously known as block gas limit), where it is assumed that block producers care about the long term health and decentralisation of the chain. Indeed, the block gas target is one of the only parameters in Ethereum that is not dictated by node consensus, but instead is chosen by block producers. This decision was initially made to allow urgent changes in the block gas target if necessary. Both drastically increasing or decreasing this parameter could have serious consequences that may not be desired. It is therefore a critical parameter that should require node consensus to avoid any sudden harmful change imposed by a small number of actors on the rest of the network.
+
+## Specification
+This EIP assumes that EIP-1559 is implemented and blocks' `gasLimit` was replaced with `gasTarget`.
+
+#### Block Header
+
+As of `FORK_BLOCK_NUMBER`, remove `gas_target` field from block headers.
```

Comment:
A nit, it is probably easier to keep the gas limit in the block and ignore than remove. But since we're adding BASEFEE already, the point may be moot.

Reply:
Indeed, I believe EIP-1559 already forces a change to the data structure to be hashed for block headers.

Reply:
1559 adds an item on the end, so if you have code that hashes the block and/or plucks items out of it by offset then 1559 won't break anything.  However, if we pull gas limit out (rather than just ignore it) any code that plucks data *after* the gas limit will break.

Reply:
Fair, then for simplicity I'll remove this aspect from the EIP. Even though it would become mainly a cosmetic field, we probably need to add a check in the consensus that the gasLimit provided in the header is equal to the hardcoded gas limit, in case people use the former. Would you both agree?